daal4py/sklearn/decomposition/_pca.py

```diff
@@ -0,0 +1,235 @@
+#
+#*******************************************************************************
+# Copyright 2014-2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#******************************************************************************/
+
+import numpy as np
+import numbers
+
+from sklearn import decomposition
+from sklearn.utils import check_array
+
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils.extmath import stable_cumsum
+
+from math import sqrt
+
+import daal4py
+from .._utils import getFPType, get_patch_message, sklearn_check_version
+import logging
+
+if sklearn_check_version('0.22'):
+    from sklearn.decomposition._pca import PCA as PCA_original
+else:
+    from sklearn.decomposition.pca import PCA as PCA_original
+
+if sklearn_check_version('0.23'):
+    from sklearn.decomposition._pca import _infer_dimension
+elif sklearn_check_version('0.22'):
+    from sklearn.decomposition._pca import _infer_dimension_
+else:
+    from sklearn.decomposition.pca import _infer_dimension_
+
+
```

Comment:
Why didn't you transfer the `_fit` method? It seems to me that you can also edit there + the information about the conformance will not be distorted.